graphene_django/converter.py

```diff
@@ -29,7 +29,10 @@
 
 
 def convert_choice_name(name):
-    name = to_const(force_text(name))
+    name = force_text(name).encode('utf8').decode('ascii', 'ignore')
+    name = to_const(name)
+    if name.startswith('_'):
+        name = "A%s" % name
```

Comment:
@zbyte64 I'm a bit confused with what is happening here. If the name starts with `_` you're prefixing it with a `A`? Why is that?

Reply:
`assert_valid_name` requires the name not to start with an `_`. Without this, a name like `_amount` would be caught and `A_` would be prefixed to become `A__amount`. With this change, the name becomes `A_amount`, removing the extra `_`

Reply:
@zbyte64 I don't think that is right. Running `assert_valid_name` on a values with both `_` and `__` work fine:

```python
In [1]: from graphql import assert_valid_name

In [2]: assert_valid_name('foo')

In [3]: assert_valid_name('_foo')

In [4]: assert_valid_name('__foo')
```

Reply:
Erm, I mispoke. `assert_valid_name`  will accept names that start in `_` but graphene's Enum will skip over them when populate members. Thus we need to prefix this special case ourselves.

Reply:
Specifically in the code where this happens: https://github.com/graphql-python/graphene/blob/6c7cd4e4fa6e29f55ebf02f173281d46d9f8fc1c/graphene/pyutils/enum.py#L109

Reply:
Oh I see. Reading through the code it looks like the logic is a bit more subtle than just ignoring values that begin with `_` though. It looks like it ignores values that are "sunder" or "dunder" only (i.e. `_amount_` or `__amount__`). For example if I construct the following Enum:

```python
In [4]: enum = Enum('foo', [('amount', 'Amount'), ('_amount', 'Amount'), ('__amount', 'Amount'), ('__amount__', 'Amount')])

# It looks like this:
In [5]: enum._meta.enum.__members__
Out[5]:
mappingproxy({'amount': <foo.amount: 'Amount'>,
              '_amount': <foo.amount: 'Amount'>,
              '__amount': <foo.amount: 'Amount'>})
```

(If I try and construct an enum with `('_amount_', 'Amount')` it just fails completely)

From a GraphQL spec point of view names that begin with `__` are not allowed because they are reserved for introspection but names that begin with `_` are allowed. So based on that I would recommend that the logic here be something like:

```python
if _is_sunder(name) or _is_dunder(name):
	name = "A%s" % name
if name.startswith('__'):
	name = "A%s" % name
```

What do you think @zbyte64 ?

Reply:
If that's the case then sunder should pass through and the prepend should only apply to dunder.

Reply:
No I don't think that's right. If a "sunder" is passed to `Enum` then it raises an exception (https://github.com/graphql-python/graphene/blob/6c7cd4e4fa6e29f55ebf02f173281d46d9f8fc1c/graphene/pyutils/enum.py#L131) which is not great. So the code should prefix "sunder" as well as "dunder". Also the code needs to catch names that just start with `__` but don't end with it.

Reply:
I think I see now. I expanded the test case to cover the various underscore mixtures.

Reply:
How does a normal Python enum handle these sunder and dunder values?

Reply:
Creating a python enum with a sunder causes:
`ValueError: _names_ are reserved for future Enum use`

Dunder is allowed but is ignored (doesn't generate a member).

Reply:
Then does it make sense to have the same limitations in the Graphene Enum type instead of tacking `A_` on the front?

Reply:
We're already converting choices that wouldn't be allowed, like those that start with a number.